### PR TITLE
Add Team To Tournament

### LINKED ISSUE #115: Add teams to tournament
What has to be done:
The frontend needs to add teams to a tournament.

Why it has to be done:
So that the team can participate to the tournament.

When the task is done:
When teams can join the tournament and this is saved in the database.

### PR DESCRIPTION
**Pull request by larsvandriel**
Closes #115

### CODE REVIEW

File: src/main/java/com/fontys/api/service/TournamentService.java [hunk 1]
```diff
@@ -37,15 +41,15 @@ public Tournament createTournament(String name, String description, Integer owne
     @Transactional
     public Tournament updateTournament(Integer id, String name, String description, Integer ownerId, Integer numberOfTeams) throws InvalidAttributeValueException {
 
-        validateTournament(id);
+        Tournament tournamentOld = validateTournament(id);
```
Comment (chf3kl9) [inline src/main/java/com/fontys/api/service/TournamentService.java]: We don't need to rename this to tournamentOld, as we don't use any 'tournamentNew' or 'tournament'

File: src/test/java/com/fontys/api/service/TournamentServiceTest.java [hunk 2]
```diff
@@ -18,18 +22,22 @@
...
-        tournamentService = new TournamentService(tournamentRepositoryMock, userRepositoryMock);
+        teamRepositoryMock = mock(TeamRepository.class);
+
+        tournamentService = new TournamentService(tournamentRepositoryMock, userRepositoryMock, teamRepositoryMock);
     }
 
     @Test
     void deleteTournamentShouldReturnDeletedString() {
-        Tournament t = new Tournament(1, "testTournament", null, new User("uuid"), 2);
+        Tournament t = new Tournament(1, "testTournament", null, new User("uuid"), 2, new ArrayList<>());
```
Comment (chf3kl9) [inline src/test/java/com/fontys/api/service/TournamentServiceTest.java]: We still have the constructor that does not require Teams, so this isn't necessary to do

File: src/test/java/com/fontys/api/service/TournamentServiceTest.java [hunk 3]
```diff
@@ -133,8 +141,8 @@ private void createTournamentTestInvalid(String name, String description, Intege
 
     private void updateTournamentTestValid() {
         User user = new User(1, "testOwner");
-        Tournament tournament = new Tournament(1, "testTournament", "description", user, 2);
-        Tournament result = new Tournament(1, "testTournamentNew", "descriptionNew", user, 4);
+        Tournament tournament = new Tournament(1, "testTournament", "description", user, 2, new ArrayList<>());
```
Comment (chf3kl9) [inline src/test/java/com/fontys/api/service/TournamentServiceTest.java]: Ditto

File: src/test/java/com/fontys/api/service/TournamentServiceTest.java [hunk 4]
```diff
@@ -159,8 +167,8 @@ private void updateTournamentTestInValid(String errorMessage) {
         User user = new User(1, "testOwner");
         User user1 = new User(2, "testOwnerNew");
 
-        Tournament tournament = new Tournament(1, "testTournament", "description", user, 2);
-        Tournament result = new Tournament(1, "testTournamentNew", "descriptionNew", user1, 4);
+        Tournament tournament = new Tournament(1, "testTournament", "description", user, 2, new ArrayList<>());
+        Tournament result = new Tournament(1, "testTournamentNew", "descriptionNew", user1, 4, new ArrayList<>());
```
Comment (chf3kl9) [inline src/test/java/com/fontys/api/service/TournamentServiceTest.java]: Ditto

File: src/main/java/com/fontys/api/service/TournamentService.java [hunk 5]
```diff
@@ -75,11 +79,43 @@ public String deleteTournament(Integer id) {
...
+        {
+            Tournament tournament1 = tournament.get();
+            Team team1 = team.get();
+            if(tournament1.getTeams().contains(team1))
+            {
+                return "The team already joined the tournament!";
+            }
+            if(tournament1.getNumberOfTeams() == tournament1.getTeams().size())
+            {
+                return "The tournament is currently filled with teams!";
```
Comment (chf3kl9) [inline src/main/java/com/fontys/api/service/TournamentService.java]: Maybe we can return an error message that is more clear.
Maybe: "The tournament no longer has empty slots for this team"

File: src/test/java/com/fontys/api/service/TournamentServiceTest.java [hunk 6]
```diff
@@ -180,4 +188,121 @@ private void updateTournamentTestInValid(String errorMessage) {
...
+        teams.add(team4);
+
+        Tournament tournament = new Tournament(1, "Tournament1", "Tournament 1", user, 4, teams);
+
+        when(tournamentRepositoryMock.findById(Mockito.anyInt())).thenReturn(Optional.of(tournament));
+        when(teamRepositoryMock.findById(Mockito.anyInt())).thenReturn(Optional.of(team5));
+
+        String response = this.tournamentService.addTeamToTournament(tournament.getId(), team5.getId());
+
+        assertEquals("The tournament is currently filled with teams!", response);
```
Comment (chf3kl9) [inline src/test/java/com/fontys/api/service/TournamentServiceTest.java]: Same as TournamentService

> Maybe we can return an error message that is more clear.
Maybe: "The tournament no longer has empty slots for this team"

**Review by chf3kl9 — CHANGES_REQUESTED** [review]

**Review by chf3kl9 — APPROVED** [review]

**Review by lfb0801 — APPROVED** [review]

**Pull request merged**